### ISSUE TITLE
Change `str::as_bytes` to hex parsing in tests for clarity

### ISSUE THREAD

**Issue by Kixunil** [rust-bitcoin/rust-bitcoin#1431]
>The potentially confusing as_bytes is still here but since it's technically correct

Oh, right, I missed checking this in the scrollback.

We should file an issue -- I've read that code multiple times and even with you pointing it out, I keep missing this because all the other tests in this crate have some hex-decoding logic somewhere.

_Originally posted by @apoelstra in https://github.com/rust-bitcoin/rust-bitcoin/issues/1412#issuecomment-1333778819_

**Comment by tcharding** [comment]
I believe this issue is stale, I looked in `script/tests.rs` and also went through the output of `gg '\.as_bytes\(\)' | grep -v script`

**Comment by Kixunil** [comment]
No, it's still there, it was just changed to `b"..."`. Try `rg "let malformed_key"`

**Comment by tcharding** [comment]
Oh yes I saw those, I figured the `b` in front makes it obvious that it is not a hex string. Hence making this issue stale.

**Comment by Kixunil** [comment]
No it doesn't. It looks human-readable and it's just one character difference.

I suggest we just hex-encode it and use `hex!` on the result.

**Comment by tcharding** [comment]
No stress
